sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -464,6 +464,41 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(StringSpace(s1), null, row2)
   }
 
+  test("RegexReplace") {
+    val row1 = create_row("100-200", "(\\d+)", "num")
```

Comment:
@rxin, I've checked the code, only the non-codegen projection will keep a single expression instance among `checkEvaluation` calls, that means only the non-codegen version can test the mutable state of the expression.

I will create another PR for updating the unit test framework itself (in the `ExpressionEvalHelper`), so we can test the mutable status for codegen expression, but this test will not be changed, can you go ahead to review this?